File: backend/internal/game/decay.go
package game

import (
	"math"
	"time"
)

// Tau is the decay time constant (REF: docs/game-design.md §Time Decay)
// port of RevKit Decay.tau
const Tau = 3 * 7 * 24 * 60 * 60 // seconds

// ExpiryThreshold is the effective score below which ownership lapses
const ExpiryThreshold = 0.5

// Expired reports whether a tile's ownership has decayed away entirely
// port of RevKit Decay.expired
func Expired(claimScore float64, lastDrivenAt, now time.Time) bool {
	return EffectiveScore(claimScore, lastDrivenAt, now) < ExpiryThreshold
}

// EffectiveScore returns claimScore * exp(-Δt/τ)
// port of RevKit's Decay.effectiveScore
func EffectiveScore(claimScore float64, lastDrivenAt, now time.Time) float64 {
	delta := now.Sub(lastDrivenAt).Seconds()
	if delta < 0 {
		delta = 0
	}
	return claimScore * math.Exp(-delta/Tau)
}

File: ios/RevKit/Sources/RevKit/Game/Decay.swift
import Foundation

/// time decay of a tile's stored claim score (REF: docs/game-design.md §Time Decay)
public enum Decay {
    /// decay time constant, τ (tau)
    public static let tau: TimeInterval = 3 * 7 * 24 * 60 * 60 // = aprox. 3 weeks

    /// `claim_score * exp(-Δt/τ)`
    public static func effectiveScore(claimScore: Double, lastDrivenAt: Date, now: Date) -> Double {
        let delta = max(0, now.timeIntervalSince(lastDrivenAt))
        return claimScore * exp(-delta / tau)
    }
}
